packages/big-design/src/managers/alerts/manager.ts

```diff
@@ -0,0 +1,108 @@
+import { AlertProps } from '../../components/Alert';
+
+import { Subscriber, TypeMap } from './types';
+
+interface PrivateAlert extends AlertProps {
+  onClose(): void;
+}
+
+class AlertsManager {
+  private alerts: PrivateAlert[] = [];
+  private counter = 0;
+  private subscribers: Subscriber[] = [];
+  private readonly typeMap: TypeMap = {
+    error: 0,
+    warning: 1,
+    success: 2,
+    info: 3,
+  };
+
+  add(alert: AlertProps, dismissCallback?: () => void): string {
+    if (alert.key && this.containsKey(alert.key)) {
+      this.remove(alert.key);
+    }
+
+    const key = this.getKey(alert.key);
+    const onClose = () => {
+      if (typeof dismissCallback === 'function') {
+        dismissCallback(); // Should we return something with this?
+      }
+
+      if (typeof alert.onClose === 'function') {
+        alert.onClose();
+      }
+
+      this.remove(key);
+    };
+
+    const newAlert: PrivateAlert = { ...alert, key, onClose };
+
+    this.alerts = this.alerts.concat([newAlert]).sort(this.sortAlerts);
+
+    this.contactSubscribers();
+
+    return key;
+  }
+
+  remove(key: string): AlertProps | undefined {
+    let removed;
+
+    this.alerts = this.alerts.reduce((acc, alert) => {
+      if (alert.key === key) {
+        removed = alert;
+
+        return acc;
+      }
+
+      acc.push(alert);
+
+      return acc;
+    }, [] as PrivateAlert[]);
+
+    this.contactSubscribers();
+
+    return removed;
+  }
+
+  subscribe(subscriber: Subscriber) {
+    this.subscribers.push(subscriber);
+  }
+
+  unsubscribe() {
+    this.alerts = [];
```

Comment:
Is there a reason we are resetting all this here? Seems to me that we only need to find the subscriber and remove it from the list of subscribers :thinking: